drivers/SmartThings/jbl/src/discovery_mdns.lua

```diff
@@ -0,0 +1,154 @@
+            return answer_item.kind.TxtRecord.text
+        end
+    end
+end
+
+local function get_srvname_by_hostname(hostname, discovery_responses)
+    for _,answer_item in pairs(discovery_responses.answers or {}) do
+        if answer_item.kind.SrvRecord ~= nil and answer_item.kind.SrvRecord.target == hostname then
+            return answer_item.name
+        end
+    end
+end
+
+local function get_hostname_by_ip(ip, discovery_responses)
+    for _,answer_item in pairs(discovery_responses.answers or {}) do
+        if answer_item.kind.ARecord ~= nil and answer_item.kind.ARecord.ipv4 == ip then
+            return answer_item.name
+        end
+    end
+end
+
+
+local function find_text_in_answers_by_ip(ip, discovery_responses)
+    local hostname = get_hostname_by_ip(ip, discovery_responses)
+    local srvname = get_srvname_by_hostname(hostname, discovery_responses)
+    local text = get_text_by_srvname(srvname,discovery_responses)
+
+    return text
+end
+
+function discovery_mdns.find_text_list_in_mdns_response(driver, ip, discovery_responses)
+    local text_list = {}
+
+    for _, found_item in pairs(discovery_responses.found or {}) do
+        if found_item.host_info.address == ip then
+            for _, raw_text_array in pairs(found_item.txt.text or {}) do
+                local text_item =  byte_array_to_plain_text(raw_text_array)
+                table.insert(text_list, text_item)
+            end
+        end
+    end
+
+    local answer_text = find_text_in_answers_by_ip(ip, discovery_responses)
+    for _, text_item in pairs(answer_text or {}) do
+        table.insert(text_list, text_item)
+    end
+    return text_list
+end
+
+local function filter_response_by_servie_name(service_type, domain, discovery_responses)
+    local filtered_responses = {
+      answers = {},
+      found = {}
+    }
+
+    for _, answer in pairs(discovery_responses.answers or {}) do
+      table.insert(filtered_responses.answers, answer)
+    end
+
+    for _, additional in pairs(discovery_responses.additional or {}) do
+      table.insert(filtered_responses.answers, additional)
+    end
+
+    for _, found in pairs(discovery_responses.found or {}) do
+      if found.service_info.service_type == service_type then
+        table.insert(filtered_responses.found, found)
+      end
+    end
+
+    return filtered_responses
+end
+
+local function insert_dni_ip_from_answers(driver, filtered_responses, target_table)
+    for _, answer in pairs(filtered_responses.answers) do
+        local dni, ip
+        log.info("answer_name, arecod = " .. tostring(answer.name) .. ", " .. tostring(answer.kind.ARecord))
+
+        if answer.kind.ARecord ~= nil then
+            ip = answer.kind.ARecord.ipv4
+        end
+
+        if ip ~= nil then
+            if dni == nil then
+                dni = driver.discovery_helper.get_dni(driver, ip, filtered_responses)
+            end
+    
+            if dni ~= nil then
+                target_table[dni] = ip
```

Comment:
I would not recommend using the IP address as the Device Network Identifier:

1. The IP can change for a variety of reasons, such as in the case that DHCP is being used on the network,
2. The Device Network ID cannot be updated after the device is created.

The Device Network ID should be both unique and stable. Typically, for a LAN device, we use the MAC address if that's available.

If the MAC isn't available, it should be some other stable/unchanging and unique value.